### PR TITLE
Move containers to file locks from c/storage

### PR DESCRIPTION
**Pull request by mheon**
This will help prevent synchronization issues. It does add a lot of open file descriptors (one per container we've read into memory), but we can look into solving that once it becomes an issue.

### CODE REVIEW

File: libpod/runtime.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ type Runtime struct {
 	storageService *storageService
 	imageContext   *types.SystemContext
 	ociRuntime     *OCIRuntime
+	locksDir       string
```
Comment (TomSweeneyRedHat) [inline libpod/runtime.go]: nit question.  You've lock and lockpath, but locksdir (plural).  I'd make this lockDir, but not a biggy.

File: libpod/sql_state_test.go [hunk 2]
```diff
@@ -102,7 +111,7 @@ func getEmptyState() (s State, p string, err error) {
 	dbPath := filepath.Join(tmpDir, "db.sql")
 	lockPath := filepath.Join(tmpDir, "db.lck")
 
-	state, err := NewSQLState(dbPath, lockPath, tmpDir, nil)
+	state, err := NewSQLState(dbPath, lockPath, tmpDir, tmpDir, nil)
```
Comment (TomSweeneyRedHat) [inline libpod/sql_state_test.go]: 'nother nit.  You'd never have your specdir and lockdir be the same right?  I woudln't test that way.

**Comment by rhatdan** [comment]
@nalind PTAL

File: libpod/runtime.go (hunk 1 above)
Comment (rhatdan) [reply libpod/runtime.go]: Yes lets do lockdir. If there is a difference between lockdir and lockpath.  Otherwise lockpath.

**Comment by mheon** [comment]
Pushed two more commits. Comments should be addressed. I've also removed the independent file locking for the SQLite DB - SQLite already has locking built in, and it's probably better tested and more durable than what I wrote.

**Comment by rhatdan** [comment]
@mheon So we still need this patch?

**Comment by mheon** [comment]
Yeah, this is to move container locking. So we got rid of database locking, and added file locks to all containers.

**Comment by mheon** [comment]
Git validation seems broken

**Comment by mheon** [comment]
@rhatdan PTAL. This has been setting for awhile, and we should make efforts to get it in.

File: libpod/sql_state_internal.go [hunk 3]
```diff
@@ -384,6 +386,19 @@ func ctrFromScannable(row scannable, runtime *Runtime, specsDir string) (*Contai
 	ctr.valid = true
 	ctr.runtime = runtime
 
+	// Ensure the lockfile exists
```
Comment (rhatdan) [inline libpod/sql_state_internal.go]: Why do we need to ensure the lockfile exists?  If it does not, thenstorage.GetLockfile will create it?

**Comment by rhatdan** [comment]
One question/nit.  Then LGTM

File: libpod/sql_state_internal.go (hunk 3 above)
Comment (mheon) [reply libpod/sql_state_internal.go]: If the lockfile doesn't exist, that means something strange has happened, as the container should have made it when it first started. But even so this is probably a pointless check. Removing.

**Comment by mheon** [comment]
Comments addressed

**Comment by mheon** [comment]
bot, retest this please

**Comment by mheon** [comment]
Pushed a trivial comments change to rekick tests

**Comment by mheon** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
Humu is not working. merging manually.

**Pull request merged**